lib/src/registry/registry.dart

```diff
@@ -76,3 +77,97 @@ class DynamicFactoryConfig extends FactoryConfig {
     return factory(algorithmName, match);
   }
 }
+
+class _RegistryImpl implements FactoryRegistry {
+  static const int _CONSTRUCTOR_CACHE_SIZE = 25;
+
+  final Map<Type, Map<String, RegistrableConstructor>> _staticFactories;
+  final Map<Type, Set<DynamicFactoryConfig>> _dynamicFactories;
+
+  final Map<String, RegistrableConstructor> _constructorCache =
+      Map<String, RegistrableConstructor>();
+
+  bool _initialized = false;
+
+  _RegistryImpl()
+      : _staticFactories = Map<Type, Map<String, RegistrableConstructor>>(),
+        _dynamicFactories = Map<Type, Set<DynamicFactoryConfig>>();
+
+  @override
+  T create<T>(String registrableName) {
+    Type type = T;
+    RegistrableConstructor constructor = getConstructor(type, registrableName);
+    T result = constructor();
+    return result;
+  }
+
+  RegistrableConstructor getConstructor(Type type, String registrableName) {
+    RegistrableConstructor constructor =
+        _constructorCache["${type}.${registrableName}"];
+
+    if (constructor == null) {
+      constructor = _createConstructor(type, registrableName);
+      if (_constructorCache.length > _CONSTRUCTOR_CACHE_SIZE) {
+        _constructorCache.clear();
+      }
+      _constructorCache["${type}.${registrableName}"] = constructor;
+    }
+    return constructor;
+  }
+
+  RegistrableConstructor _createConstructor(Type type, String registrableName) {
+    // Init lazily
+    _checkInit();
+
+    if (_staticFactories.containsKey(type) &&
+        _staticFactories[type].containsKey(registrableName)) {
+      return _staticFactories[type][registrableName];
+    }
+
+    if (_dynamicFactories.containsKey(type)) {
+      for (DynamicFactoryConfig factory in _dynamicFactories[type]) {
+        RegistrableConstructor constructor =
+            factory.tryFactory(registrableName);
+        if (constructor != null) {
+          return constructor;
+        }
+      }
+    }
+
+    // No factory found
+    throw new RegistryFactoryException.unknown(registrableName, type);
+  }
+
+  void _checkInit() {
+    if (!_initialized) {
+      _initialize();
+    }
+  }
+
+  @override
+  void register<T>(FactoryConfig config) {
+    Type t = T;
+    if (config is StaticFactoryConfig) {
+      _addStaticFactoryConfig(config);
+    } else if (config is DynamicFactoryConfig) {
+      _addDynamicFactoryConfig(config);
+    }
+  }
+
+  void _addStaticFactoryConfig(StaticFactoryConfig config) {
+    Map factories = _staticFactories.putIfAbsent(
+        config.type, () => Map<String, RegistrableConstructor>());
+    factories[config.algorithmName] = config.factory;
+  }
+
+  void _addDynamicFactoryConfig(DynamicFactoryConfig config) {
+    Set factories = _dynamicFactories.putIfAbsent(
+        config.type, () => Set<DynamicFactoryConfig>());
+    factories.add(config);
+  }
+
+  void _initialize() {
+    registerFactories(this);
```

Comment:
here we call a method to register all `FactoryConfigs`.
Actual implementation is on [another file](https://github.com/PointyCastle/pointycastle/pull/148/files#diff-fd5218667fabd439f3bc1bd3ca2274ec).